src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1252,6 +1252,18 @@
   (jdbc/do-commands
     "create index resource_params_cache_parameters_idx on resource_params_cache using gin (parameters)"))
 
+(defn fix-missing-edges-fk-constraint []
+  (when-not (sutils/constraint-exists? "certnames" "edges_certname_fkey")
+    (log/info "Cleaning up orphaned edges")
+    (jdbc/delete! :edges ["certname not in (select certname from certnames)"])
```

Comment:
That might all need to be done inside of a transaction.  I'm not sure if the migrations block out incoming traffic or not.